AutoRest/Generators/Ruby/Azure.Ruby/AzureRubyCodeGenerator.cs

```diff
@@ -58,6 +66,76 @@ public override void NormalizeClientModel(ServiceClient serviceClient)
             AzureExtensions.NormalizeAzureClientModel(serviceClient, Settings, CodeNamer);
             CorrectFilterParameters(serviceClient);
             base.NormalizeClientModel(serviceClient);
+            AddRubyPageableMethod(serviceClient);
+            ApplyPagination(serviceClient);
+
+        }
+
+        /// <summary>
+        /// Adds method to use for autopagination.
+        /// </summary>
+        /// <param name="serviceClient">The service client.</param>
+        private static void AddRubyPageableMethod(ServiceClient serviceClient)
+        {
+            if (serviceClient == null)
+            {
+                throw new ArgumentNullException(nameof(serviceClient));
+            }
+
+            for (int i = 0; i < serviceClient.Methods.Count; i++)
+            {
+                var method = serviceClient.Methods[i];
+                if (method.Extensions.ContainsKey(AzureExtensions.PageableExtension))
+                {
+                    var pageableExtension = JsonConvert.DeserializeObject<Model.PageableExtension>(method.Extensions[AzureExtensions.PageableExtension].ToString());
+                    if (pageableExtension != null && !method.Extensions.ContainsKey("nextLinkMethod") && !string.IsNullOrWhiteSpace(pageableExtension.NextLinkName))
+                    {
+                        serviceClient.Methods.Insert(i, (Method)method.Clone());
+                        i++;
+                    }
+                    serviceClient.Methods[i].Extensions.Remove(AzureExtensions.PageableExtension);
```

Comment:
should this line (serviceClient.Methods[i].Extensions.Remove(AzureExtensions.PageableExtension);) be inside if condition?